python/cudf_polars/cudf_polars/experimental/distinct.py

```diff
@@ -141,13 +141,16 @@ def lower_distinct(
             partitioned_on=shuffle_keys,
         )
 
+    # new_node always comes from `ir.reconstruct`, which has type Distinct
+    assert isinstance(new_node, Distinct)
```

Comment:
`mypy` wasn't able to figure this out on its own, but AFAICT (and the tests confirm that) this is correct.